src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -0,0 +1,67 @@
+package lotto.domain;
+
+import static lotto.common.exception.ErrorMessage.ERROR_NUMBER_RANGE;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+    public static final int LOTTO_RANGE_MINIMUM = 1;
+    public static final int LOTTO_RANGE_MAXIMUM = 45;
+
+    private static final Map<Integer, LottoNumber> CACHE = new HashMap<>();
+
+    private final int number;
+
+    private LottoNumber(int number) {
+        this.number = number;
+    }
+
+    public static LottoNumber of(int number) {
+        validate(number);
+
+        return CACHE.computeIfAbsent(number, LottoNumber::new);
+    }
```

Comment:
```suggestion
    private LottoNumber(int number) {
        this.number = number;
        validate(number);
    }

    public static LottoNumber of(int number) {
        return CACHE.computeIfAbsent(number, LottoNumber::new);
    }
```
팩토리 메서드가 여럿 있을 때 메서드마다 다른 검증로직을 적용할 것이 아니라면, 객체의 생성자에 배치하는 것을 권장드려요